redpanda-otel-exporter/log_exporter.go

```diff
@@ -0,0 +1,229 @@
+//
+// Required: WithBrokers option must be provided.
+//
+// Example:
+//
+//	exporter, err := NewLogExporter(
+//	    WithBrokers("localhost:9092"),
+//	    WithResource(res),
+//	)
+//
+// To use a custom topic:
+//
+//	exporter, err := NewLogExporter(
+//	    WithBrokers("localhost:9092"),
+//	    WithTopic("custom-logs"),
+//	)
+func NewLogExporter(opts ...Option) (*LogExporter, error) {
+	exporter, err := newExporter("otel-logs", opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &LogExporter{
+		Exporter: exporter,
+	}, nil
+}
+
+// Export exports a batch of log records to Kafka.
+func (e *LogExporter) Export(ctx context.Context, records []sdklog.Record) error {
+	if len(records) == 0 {
+		return nil
+	}
+
+	if e.config.serializationFormat == SerializationFormatProtobuf {
+		return e.exportLogsProtobuf(ctx, records)
+	}
+	return e.exportLogsJSON(ctx, records)
+}
+
+// exportLogsJSON exports logs in JSON format
+func (e *LogExporter) exportLogsJSON(ctx context.Context, records []sdklog.Record) error {
+	kafkaRecords := make([]*kgo.Record, 0, len(records))
+
+	for _, record := range records {
+		logData := e.logRecordToMap(record)
+
+		value, err := marshalJSON(logData)
+		if err != nil {
+			return fmt.Errorf("failed to marshal log record: %w", err)
+		}
+
+		// Use a combination of severity and timestamp as key for partitioning
+		key := fmt.Appendf(nil, "%s-%d", record.Severity().String(), record.Timestamp().UnixNano())
+
+		kafkaRecord := &kgo.Record{
+			Topic: e.config.topic,
+			Key:   key,
+			Value: value,
+		}
+
+		// Add resource attributes as Kafka headers instead of in the JSON body
+		if e.config.resource != nil {
+			kafkaRecord.Headers = resourceToHeaders(e.config.resource)
+		}
+
+		kafkaRecords = append(kafkaRecords, kafkaRecord)
+	}
+
+	return e.produceBatch(ctx, kafkaRecords)
+}
+
+// exportLogsProtobuf exports logs in OTLP protobuf format
+func (e *LogExporter) exportLogsProtobuf(ctx context.Context, records []sdklog.Record) error {
+	if len(records) == 0 {
+		return nil
+	}
+
+	kafkaRecords := make([]*kgo.Record, 0, len(records))
+
+	for _, record := range records {
+		// Create a resource logs with a single log record
+		// This maintains consistency with JSON format (one record per log)
+		resourceLogs := &logspb.ResourceLogs{
+			Resource: resourceToProto(e.config.resource),
+			ScopeLogs: []*logspb.ScopeLogs{
+				{
+					Scope:      instrumentationScopeToProto(record.InstrumentationScope()),
+					LogRecords: []*logspb.LogRecord{logRecordToProto(record)},
+				},
+			},
+		}
+
+		value, err := marshalProtobuf(resourceLogs)
+		if err != nil {
+			return fmt.Errorf("failed to marshal log record to protobuf: %w", err)
+		}
+
+		// Use a combination of severity and timestamp as key for partitioning (same as JSON format)
+		key := fmt.Appendf(nil, "%s-%d", record.Severity().String(), record.Timestamp().UnixNano())
```

Comment:
Wondering what the purpose of the key here is? Do we want some partitioning - based on the key I guess we don't?